src/client/datascience/data-viewing/data-wrangler/dataWrangler.ts

```diff
@@ -0,0 +1,821 @@
+            localize.DataScience.dataWranglerTitle(),
+            PREFERRED_VIEWGROUP,
+            ViewColumn.One
+        );
+        this.commands.set(DataWranglerCommands.Describe, this.getColumnStats.bind(this));
+        this.commands.set(DataWranglerCommands.ExportToPythonScript, this.generatePythonCode.bind(this));
+        this.commands.set(DataWranglerCommands.ExportToNotebook, this.generateNotebook.bind(this));
+        this.commands.set(DataWranglerCommands.RenameColumn, this.renameColumn.bind(this));
+        this.commands.set(DataWranglerCommands.Drop, this.drop.bind(this));
+        this.commands.set(DataWranglerCommands.DropDuplicates, this.dropDuplicates.bind(this));
+        this.commands.set(DataWranglerCommands.DropNa, this.dropNa.bind(this));
+        this.commands.set(DataWranglerCommands.NormalizeColumn, this.normalizeColumn.bind(this));
+        this.commands.set(DataWranglerCommands.FillNa, this.fillNa.bind(this));
+        this.commands.set(DataWranglerCommands.GetHistoryItem, this.getHistoryItem.bind(this));
+        this.commands.set(DataWranglerCommands.CoerceColumn, this.coerceColumn.bind(this));
+        this.commands.set(DataWranglerCommands.ReplaceAllColumn, this.replaceAllColumn.bind(this));
+        this.commands.set(DataWranglerCommands.RemoveHistoryItem, this.removeHistoryItem.bind(this));
+        this.commands.set(DataWranglerCommands.ExportToCsv, this.exportToCsv.bind(this));
+        this.commands.set(DataWranglerCommands.RespondToPreview, this.respondToPreview.bind(this));
+
+        this.onDidDispose(this.dataWranglerDisposed, this);
+    }
+
+    public async showData(
+        dataProvider: IDataViewerDataProvider,
+        title: string,
+        webviewPanel?: WebviewPanel
+    ): Promise<void> {
+        if (!this.isDisposed) {
+            // Save the data provider
+            this.dataProvider = dataProvider;
+
+            // Load the web panel using our current directory as we don't expect to load any other files
+            await super.loadWebview(process.cwd(), webviewPanel).catch(traceError);
+            const settings = this.configService.getSettings();
+            if (settings && settings.dataWrangler && settings.dataWrangler.sidePanelSections) {
+                const wantedPanels = settings.dataWrangler.sidePanelSections;
+                this.postMessage(
+                    DataWranglerMessages.SetSidePanels,
+                    wantedPanels as SidePanelSections[]
+                ).ignoreErrors();
+            }
+
+            // Use Data Viewer logic to show initial data
+            const dataFrameInfo = await this.showInitialData(title);
+            this.sourceFile = dataFrameInfo.sourceFile;
+
+            this.historyList.push({
+                description: DataScience.dataWranglerImportDescription(),
+                code: `import pandas as pd\r\ndf = pd.read_csv(r'${this.sourceFile ?? ''}')\n`,
```

Comment:
nit: can we have a helper for multiline codegen e.g. something like
```
createCode([
  "import pandas as pd",
  `df = pd.read_csv(r'${this.sourceFile ?? '''}')`
], { lineEnding: <override default?> })
```

Reply:
I think we should probably default to the system line ending, ie from `TextDocument.eol`